pennylane_qiskit/qiskit_device2.py

```diff
@@ -623,18 +624,21 @@ def _execute_sampler(self, circuit, session):
     def _execute_estimator(self, circuit, session):
         # the Estimator primitive takes care of diagonalization and measurements itself,
         # so diagonalizing gates and measurements are not included in the circuit
-        qcirc = circuit_to_qiskit(circuit, self.num_wires, diagonalize=False, measure=False)
+        qcirc = [circuit_to_qiskit(circuit, self.num_wires, diagonalize=False, measure=False)]
         if circuit.shots:
             self.options.execution.shots = circuit.shots.total_shots
         estimator = Estimator(session=session, options=self.options)
+        compiled_circuits = self.compile_circuits(qcirc)
```

Comment:
I think I have asked this before, but to confirm that this is not the PR that we are compiling based on the device? 🤔

Reply:
I think I gave you the wrong answer last time 😅 . 

When you run compile_circuits, it compiles based on the device backend.

Reply:
In further detail:

I was confused and thought we didn't transpile when `use_primitives=False`. But actually we were transpiling when `use_primitives=False` and **not** transpiling when` use_primitives = True`. This PR just makes it so when `use_primitives=True`, we transpile.